pandas/core/base.py

```diff
@@ -632,6 +637,10 @@ def shape(self):
         """
         return self._values.shape
 
+    def __len__(self) -> int:
```

Comment:
pandas/core/base.py:1343: error: Argument 1 to "len" has incompatible type "IndexOpsMixin"; expected "Sized"

Reply:
It *might* make sense to just ignore that line; I think mypy generally doesn't play nice with Mixins so may be a few things like this, but I would imagine the intent is for that to be defined by whatever gets composed with this class

Not a big deal for me either way maybe just a note for other reviewers. Otherwise this lgtm

Reply:
@simonjayhawkins thoughts here?

Reply:
perhaps for consistency with addition for `_values` above, `__len__` could raise `AbstractMethodError` instead?

another alternative could be to add a type declaration at the top of the class instead,
`__len__: Callable[..., int]`

or ignore as @WillAyd suggested.